### DISCUSSION TITLE
useJsonForms() is giving just selected node data not complete data

### DISCUSSION THREAD

**Discussion by kuldeeparora89** [eclipse-emfcloud/emfcloud#176] [Theia Tree Editor]
Hi,

I am writing a custom enum cell renderer to dynamically create options from data present in another part of JSON.
But the problem is that I am not able to extract complete data using useJsonForms() as it is just giving the data of the selected node only.

Please suggest.

Code -

```
export const SelectTypeEnumCell = (props: EnumCellProps & VanillaRendererProps) => {
  const { data, className, id, enabled, uischema, path, handleChange } = props;
  
  const getOptions=()=>{
  const ctx = useJsonForms();
  const wholeDataObject = ctx.core?.data;
  const typeOptions:EnumOption[] = wholeDataObject.types
    ?.map(
      (type: { "@id": any }) => {
        return { label: type["@id"], value: type["@id"]} as EnumOption;
      }
    )
    ?? [];

    return   typeOptions;
  }
  return (
    <select
      className={className}
      id={id}
      disabled={!enabled}
      autoFocus={uischema.options && uischema.options.focus}
      value={data || ''}
      onChange={ev => handleChange(path, ev.target.value)}
    >
      {
        [<option value='' key={'empty'} />]
          .concat(
            getOptions().map(optionValue =>
            (
              <option value={optionValue.value} label={optionValue.label} key={optionValue.value} />
            )
            )
          )}
    </select>
  );
};
```

**Comment by kuldeeparora89** [comment]
As a workaround, we are passing the complete data using React context by extending the FormWidget.

**Reply by sdirix** [reply]
Hi @kuldeeparora89,

this is correct. JSON Forms is only invoked on the specific subset of the data, not the "whole" data. Therefore JSON Forms can only give you the access to the specific data it was handed over via `useJsonForms()`.

The solution with a custom React context which is consumed in your custom renderer is the typical approach when you need access to some data which is not managed by JSON Forms itself, so on a first glance you're doing everything right!
